src/sagemaker/model.py

```diff
@@ -215,6 +215,10 @@ def _compilation_job_config(
         job_name,
         framework,
         tags,
+        target_platform_os=None,
+        target_platform_arch=None,
+        target_platform_accelerator=None,
+        compiler_options=None,
```

Comment:
Maybe we want to have the doc strings here in `model.compile`. Need to point out that the target_instance_family param and the target_platform_* params are not usable together.